toolkit/scripts/toolchain/build_official_toolchain_rpms.sh

```diff
@@ -242,24 +244,25 @@ build_rpm_in_chroot_no_install () {
         fi
     else
         echo only building RPM $1 within the chroot
-        srpmName=$(rpmspec -q $specPath --srpm --define="with_check 1" --define="_sourcedir $specDir" --define="dist $PARAM_DIST_TAG" --queryformat %{NAME}-%{VERSION}-%{RELEASE}.src.rpm)
+        builtRpms="$(rpmspec -q $specPath "${defines[@]}" --queryformat="%{nvra}.rpm\n")"
+        srpmName=$(rpmspec -q $specPath --srpm "${defines[@]}" --queryformat %{NAME}-%{VERSION}-%{RELEASE}.src.rpm)
         srpmPath=$MARINER_INPUT_SRPMS_DIR/$srpmName
         cp $srpmPath $CHROOT_SRPMS_DIR
         chroot_and_run_rpmbuild $srpmName 2>&1 | awk '{ print strftime("time=\"%Y-%m-%dT%T%Z\""), $0; fflush(); }' | tee $TOOLCHAIN_LOGS/$srpmName.log
-        cp $CHROOT_RPMS_DIR_ARCH/$1* $FINISHED_RPM_DIR
-        cp $CHROOT_RPMS_DIR_NOARCH/$1* $FINISHED_RPM_DIR
+        copy_built_rpms $builtRpms
         cp $srpmPath $MARINER_OUTPUT_SRPMS_DIR
         echo NOT installing the package $srpmName
     fi
     echo "$1" >> $TOOLCHAIN_BUILD_LIST
     stop_record_timestamp "build packages/build/$1"
 }
 
-# Copy RPM subpackages that have a different prefix
-copy_rpm_subpackage () {
-    echo cache $1 RPMS
-    cp $CHROOT_RPMS_DIR_ARCH/$1* $FINISHED_RPM_DIR
-    cp $CHROOT_RPMS_DIR_NOARCH/$1* $FINISHED_RPM_DIR
+# Log the built RPMs and copy them to the finished RPMs directory.
+copy_built_rpms () {
+    for builtRpm in "$@"; do
+        find "$CHROOT_RPMS_DIR" -name "$builtRpm" -exec cp {} "$FINISHED_RPM_DIR" \;
+        echo "$builtRpm" >> "$TOOLCHAIN_BUILT_RPMS_LIST"
```

Comment:
Not sure if this is critical since we will fail toolchain validation if we get unexpected output, but this should only record the rpms we actually mange to find.
```suggestion
        if [[ -n "$(find "$CHROOT_RPMS_DIR" -name "$builtRpm" -exec cp {} "$FINISHED_RPM_DIR" \; -print)" ]]; then
            echo "$builtRpm" >> "$TOOLCHAIN_BUILT_RPMS_LIST"
        fi
```

Reply:
Good catch. I'd even treat this as an unexpected behaviour and error out early. This should make it easier to find the culprit.